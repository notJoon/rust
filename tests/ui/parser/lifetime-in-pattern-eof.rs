// Test case for issue #134061 - exact reproduction case
// Tests the specific case where EOF is encountered after lifetime in pattern

const x: () = |&'a
//~^ ERROR unexpected lifetime `'a` in pattern
//~| HELP remove the lifetime
//~| ERROR expected parameter name, found `<eof>`
