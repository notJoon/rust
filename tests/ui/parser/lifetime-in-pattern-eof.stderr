error: unexpected lifetime `'a` in pattern
  --> $DIR/lifetime-in-pattern-eof.rs:4:17
   |
LL | const x: () = |&'a
   |                 ^^
   |
help: remove the lifetime
   |
LL - const x: () = |&'a
LL + const x: () = |&
   |

error: expected parameter name, found `<eof>`
  --> $DIR/lifetime-in-pattern-eof.rs:4:19
   |
LL | const x: () = |&'a
   |                   ^ expected parameter name

error: aborting due to 2 previous errors

